lib/dradis/plugins/pentera/importer.rb

```diff
@@ -0,0 +1,75 @@
+module Dradis::Plugins::Pentera
+  class Importer < Dradis::Plugins::Upload::Importer
+    def self.templates
+      { evidence: 'evidence', issue: 'issue' }
+    end
+
+    # The framework will call this function if the user selects this plugin from
+    # the dropdown list and uploads a file.
+    # @returns true if the operation was successful, false otherwise
+    def import(params = {})
+      file_content = File.read(params[:file])
+
+      # Parse the uploaded file into a Ruby Hash
+      logger.info { "Parsing Pentera output from #{ params[:file] }..." }
+      @data = MultiJson.decode(file_content)
+      logger.info { 'Done.' }
+
+      @node_cache = {}
+      parse_hosts
+      parse_vulnerabilities
+    end
+
+    private
+
+    def parse_hosts
+      @data['hosts'].each do |host|
+        host_label = host['hostname'] || host['ip']
+        logger.info { "\tHost: #{host_label}" }
+
+        node = content_service.create_node(label: host_label)
+
+        node.set_property(:ip, host['ip']) if host['ip']
+        node.set_property(:hostname, host['hostname']) if host['hostname']
+        node.set_property(:os, host['hostname']) if host['hostname']
+        node.set_property(:identifier, host['id']) if host['id']
+
+        if host['domain']
+          node.set_property(:domain_name, host['domain']['name'])
+          node.set_property(:fqdn, host['domain']['fqdn'])
+          node.set_property(:netbios, host['domain']['netBIOS'])
+        end
+
+        parse_services(node, host['services'])
+
+        node.save
+        @node_cache[host['id']] = node
+      end
+    end
+
+    def parse_services(node, services)
+      services.each do |service|
+        node.set_service({
+          name: service['name'],
+          port: service['port'],
+          protocol: service['transport'],
+          state: service['state'],
+          source: :pentera
+        })
+      end
+    end
+
+    def parse_vulnerabilities
+      @data['vulnerabilities'].each do |vulnerability|
+        issue_text = template_service.process_template(template: 'issue', data: vulnerability)
+        issue = content_service.create_issue(text: issue_text, id: vulnerability['id'])
+
+        node = @node_cache[vulnerability['target_id']]
```

Comment:
In the sample file, some of the 'target' values in a vulnerability are 'Domain' rather than 'Host'. In that case, the `target_id` doesn't match up with a node ('Host'). So if the vulnerability's target is 'Domain', we won't be importing evidence for that issue. Is this intended?

Reply:
Good point. I'm not sure what to make of this data. Let's wait for the user feedback to be sure.